client/app/components/queries/ScheduleDialog.jsx

```diff
@@ -158,104 +127,96 @@ class ScheduleDialog extends React.Component {
-    const timeParts = newSchedule.time ? scheduleInLocalTime(newSchedule.time).split(':') : null;
+    newSchedule.interval = newInterval
+      ? intervalToSeconds(Number(this.state.count), newInterval)
+      : null;
+
+    const [hour, minute] = newSchedule.time ?
+      localizeTime(newSchedule.time).split(':')
+      : [null, null];
+
     this.setState({
       interval: newInterval,
       count: newInterval !== IntervalEnum.NEVER ? this.state.count : '1',
-      hour: timeParts ? timeParts[0] : null,
-      minute: timeParts ? timeParts[1] : null,
-      dayOfWeek: newSchedule.day_of_week ? WEEKDAYS_SHORT[WEEKDAYS_FULL.indexOf(newSchedule.day_of_week)] : null,
+      hour,
+      minute,
+      dayOfWeek: newSchedule.day_of_week
+        ? WEEKDAYS_SHORT[WEEKDAYS_FULL.indexOf(newSchedule.day_of_week)]
+        : null,
     });
 
-    this.props.updateQuery({
-      schedule: Object.assign(newSchedule, { interval: totalSeconds }),
-    });
+    this.newSchedule = newSchedule;
   };
+
   setCount = (e) => {
     const newCount = e.target.value;
     const totalSeconds = intervalToSeconds(parseInt(newCount, 10), this.state.interval);
     this.setState({ count: newCount });
-
-    this.props.updateQuery({
-      schedule: Object.assign({}, this.props.query.schedule, { interval: totalSeconds }),
-    });
+    this.newSchedule = { interval: totalSeconds };
   };
 
   setScheduleUntil = (momentDate, date) => {
-    this.props.updateQuery({
-      schedule: Object.assign({}, this.props.query.schedule, { until: date }),
-    });
+    this.newSchedule = { until: date };
   };
 
   setWeekday = (e) => {
     const dayOfWeek = e.target.value;
     this.setState({ dayOfWeek });
-    this.props.updateQuery({
-      schedule: Object.assign({}, this.props.query.schedule, {
-        day_of_week: dayOfWeek ? WEEKDAYS_FULL[WEEKDAYS_SHORT.indexOf(dayOfWeek)] : null,
-      }),
-    });
+    this.newSchedule = {
+      day_of_week: dayOfWeek ? WEEKDAYS_FULL[WEEKDAYS_SHORT.indexOf(dayOfWeek)] : null,
+    };
   };
 
   render() {
-    const schedule = this.props.query.schedule;
-    const format = 'YYYY-MM-DD';
-    const additionalAttributes = {};
+    const {
+      interval, minute, hour, until, count,
+    } = this.state;
 
     return (
       <Modal
         title="Refresh Schedule"
         className="schedule"
         visible={this.props.show}
         onCancel={this.props.onClose}
-        footer={[null, null]}
+        footer={null}
       >
         <div className="schedule-component">
           <div>Refresh every</div>
-          {schedule.interval ? (
-            <select value={this.state.count} onChange={this.setCount}>
-              {this.getAcceptableCounts().map(count => (
-                <option value={String(count)} key={count}>
-                  {String(count)}
-                </option>
+          {interval !== IntervalEnum.NEVER ? (
+            <select value={count} onChange={this.setCount}>
```

Comment:
How about using Ant's `Select` component instead of plain `<select>` tag?

Reply:
That's the next PR :P